src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/VSInternalCompletionItemExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis.Razor.Completion;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Completion
+{
+    internal static class VSInternalCompletionItemExtensions
+    {
+        private const string ResultIdKey = "_resultId";
+
+        public static VSInternalCompletionItem CreateWithCompletionListResultId(
```

Comment:
These two are basically pre-existing copies from the old CompletionItemExtensions variant. The last `UseCommitCharactersFrom` is the only new one